### DISCUSSION TITLE
QuiltMC

### DISCUSSION THREAD

**Discussion by ghost** [maruohon/litematica#598] [Ideas]
Port litematica to [QuiltMC](https://quiltmc.org/)
IDK if this is possible since litematica does complex stuff compare to the average mods

https://quiltmc.org/en/about/faq-devs/

**Comment by maruohon** [comment]
Can you clarify, is something not working with the Fabric mod versions under Quilt? I can at least launch the game and log in to a single player world with them while using the Quilt loader.

![2022-10-27_17 34 57](https://user-images.githubusercontent.com/2331050/198321484-327e841e-c1a1-4178-b042-409922b7b7b2.png)

**Reply by ghost** [reply]
Oh sorry for the confusion.
I haven't tried QuiltMC + Litematica +  Malilib. I thought it woudn't work because Litematica and Malilib uses Fabric libraries.

**Reply by ghost** [reply]
So is Litematic and Malilib ported to Quilt APIs?
Or is it currently using the backwards compatibility API?

**Reply by maruohon** [reply]
No they aren't ported to Quilt. Quilt is just able to load Fabric mods, at least for now. I have no idea how many of the Fabric mods in general work under Quilt and what the most common incompatibilities are. So I also don't know how well my mods actually work and if there are things that are broken or would crash, as I haven't properly tested them (i.e. tested all the features in the mods to see if they work).

But also my mods actually use *very few* actual Fabric APIs. In fact the client mods don't use anything from the  Fabric API mod, except that malilib embeds the resource loader module so that the mod resources (localizations and textures) get loaded and work. And due to this I *think* the only things that would break something in my mods under Quilt is if there are some mapping bugs/issues or if Quilt's Mixin library would work differently somehow.

**Reply by ghost** [reply]
Thank you for your clarification.
> But also my mods actually use very few actual Fabric APIs. In fact the client mods don't use anything from the Fabric API mod, except that malilib embeds the resource loader module so that the mod resources (localizations and textures) get loaded and work.

I thought using the backwards compatibility API provided by Quilt will have an effect on the performance. Since the mods use few Fabric APIs that may not be a concern (and so far I haven't noticed anything being laggy). 

Do you have any plans to port Litematica Malilib in the future? I'm not sure if this will have benefit on the project, or how hard/easy this will be.

**Reply by maruohon** [reply]
There are a few Mixins in my mods that would benefit from the more powerful bytecode editing library that Quilt has. So I probably will make dedicated Quilt ports of them at some point, but no idea when. As long as the Fabric versions work as-is, I'm in no hurry to make separate Quilt versions, as that would just increase my maintenance overhead and time required.
